source/extensions/filters/listener/tls_inspector/tls_inspector.cc

```diff
@@ -121,6 +133,11 @@ Network::FilterStatus Filter::onAccept(Network::ListenerFilterCallbacks& cb) {
   NOT_REACHED_GCOVR_EXCL_LINE;
 }
 
+bool Filter::shouldInspect(uint32_t port) const {
+  const auto& whitelist = config_->inspected_ports();
+  return whitelist.empty() || (whitelist.count(port) > 0);
```

Comment:
if you decide to stick to `unordered_set` then remove the` empty() `check and use` find()` instead of `count()`. It's faster.

Reply:
I think the `empty()` check is unavoidable. We need to enable the TLS inspector if the user does not specify any whitelisted ports.